### PR TITLE
Unbreak build with Rust 1.25

### LINKED ISSUE #298: Random segfaults (archlinux)
BUILDS
GCC: 7.2.0

Clang: 5.0.0

https://p.teknik.io/cRHo7

### PR DESCRIPTION
**Pull request by jbeich**
Fixes #298, #473, #477, #492.

### CODE REVIEW

**Comment by jbeich** [comment]
Let's keep servo/rust-toolchain removed to avoid having to backport http://github.com/mozilla/gecko-dev/commit/118c6485b67b
Rebasing 89ce888c7e30 into 491dc73e2e9e

File: config/rules.mk [hunk 1]
```diff
@@ -862,6 +862,16 @@ cargo_build_flags += --color=always
 endif
 endif
 
+# These flags are passed via `cargo rustc` and only apply to the final rustc
+# invocation (i.e., only the top-level crate, not its dependencies).
+cargo_rustc_flags = $(CARGO_RUSTCFLAGS)
+ifndef DEVELOPER_OPTIONS
+ifndef MOZ_DEBUG_RUST
+# Enable link-time optimization for release builds.
+cargo_rustc_flags += -C lto
```
Comment (jbeich) [inline config/rules.mk:871]: On Rust 1.25+ build time can be improved via `-C lto=thin -C codegen-units=8`.

**Comment by MrAlex94** [comment]
Thanks once again. Will review, test and if all works well, commit :-)

File: .mozconfig [hunk 2]
```diff
@@ -48,6 +48,8 @@ ac_add_options --disable-tests
 ac_add_options --disable-verify-mar
 
 ac_add_options --enable-application=browser
+ac_add_options --enable-release
+ac_add_options --enable-rust-simd # on x86 requires SSE2
```
Comment (jbeich) [inline .mozconfig:52]: `-C target-cpu=native` maybe broken for Rust >= 1.24, see [bug 1433747](https://bugzilla.mozilla.org/show_bug.cgi?id=1433747). A corner case unlikely to affect downstream.

**Review by hawkeye116477 — APPROVED** [review]
Works perfectly, but with Rust 1.24.1 compiling is much slower than with Rust 1.17.

**Pull request merged**